### ISSUE TITLE
A name for the new VSP staking process

### ISSUE THREAD

**Issue by xaur** [decred/dcrdocs#1156]
Since the initial [announcement](https://blog.decred.org/2020/06/02/A-More-Private-Way-to-Stake/) and release of [vspd](https://github.com/decred/vspd) and its support in dcrwallet and Decrediton, I've seen a bunch of ways how people refer to this new process:

- `vspd` (after the server implementation)
- `new mode` (e.g. https://github.com/decred/decrediton/issues/3181)
- `new VSP`
- `V3`, e.g. Decrediton code uses names like `STARTTICKETBUYERV3_*`, and `runTicketBuyerNew` in [code](https://github.com/decred/decrediton/pull/3182/files)
- and sometimes people/docs/UI just say `VSP` without specifying whether it is old or new

At the same time, it seems that the old way is called `legacy` by most people without significant confusion (with occasional use of `old`, `V2` and `dcrstakepool`).

Before it gets more chaotic, I suggest that we settle on clear terms to refer to old and new VSP staking. I have no preference whether to come up with a new term or use an existing one (e.g. `vspd`).

To do that, let's first consider in which contexts the term will be used. I think it is most needed for less technical UIs, docs, writing and support comms.

> \- my ticket is not voting, please help. Using this VSP...   
> \- sure, is it a legacy or {term} ticket?

> \- is there a way to buy a fraction of a ticket?
> \- yes, it is called "ticket splitting", but it is not supported with the new {term} mode so you'll have to use the legacy mode

> \- can I still use legacy VSPs?
> \- yes but migrating to {term} is recommended, read more here

> we had some bugs with {term} staking initially, but it is mostly fleshed out now

> I switched to {term} staking yesterday and it feels so good, thanks Decred

> In Decred you can stake either by running your own voting servers ("solo") or using the so-called Voting Service Providers (VSP). The latter has received a significant upgrade with v1.6 release. Now stakers have a choice to migrate to the {term} protocol that offers great advantages to privacy and ease of use.

Looking at the above examples, the term will be used together with the following words:

- _tickets_ purchased this way
- capable _VSPs_
- staking (ticket purchasing) _mode_ in dcrwallet/Decrediton
- and possibly _fees_ paid for such tickets

**Comment by xaur** [comment]
As part of this issue let's explain "legacy" and "vspd" tickets/VSP/staking in the Glossary.

**Comment by ukane-philemon** [comment]
@xaur Legacy vspd and new vspd in referring to old and new vspd sounds better?

**Comment by xaur** [comment]
@ukane-philemon, `legacy vspd` is confusing because previous server implementation was called `dcrstakepool` and not `vspd`. Also, word `legacy` is relative (to the moment of writing) and not future proof. When we replace vspd with something else, what we call `legacy` now will become... `legacy legacy`? :)

`new vspd` is also confusing because there is no `old vspd` currently. Also, `new` is not future proof again, what is `new` today will become `legacy` in 3 years.

**Comment by xaur** [comment]
One way to come up with short and future proof terminology for VSP staking is to add protocol version. Protocol version dictates how the ticket is handled on both server and client.

How about we just use `VSP2` to refer to dcrstakepool stuff and `VSP3` to the vspd stuff, and drop all the relative words like `legacy`, `old` and `new`?

In the use cases from my [original comment](https://github.com/decred/dcrdocs/issues/1156#issue-823141210) I think the terms do the job:

- _I have 2 VSP2 **tickets** and 8 VSP3 tickets, gonna finish migrating soon._
- _I bought 3 VSP3 **tickets** in Decrediton and have this error in the log: ..._
- _There was no good way to split VSP3 **tickets** but in the VSP4 **protocol** it will be easy to implement._
- _Hey do you still run those VSP2 **servers**?_
- _Welcome the new VSP3 **provider**!_
- _I'm getting this odd error when my dcrwallet is trying to pay VSP3 **fees**: ..._

This is short (4 characters), future-proof (when `VSP4` appears the other two will not change their meaning, unlike `legacy` and `new`), and also differentiates from `solo` tickets/staking/etc.

What do you think @jholdstock?

**Comment by ukane-philemon** [comment]
Protocol versioning sounds better.

**Comment by jholdstock** [comment]
Sounds like a good idea in theory, but not sure its worthwhile updating everything everywhere. The migration is practically complete now, only 0.3% of tickets on legacy stakepools. We are pretty much at the point where we can stop referring to them altogether.

**Comment by xaur** [comment]
Good point. Things happened: most operators have upgraded to vspd, most tickets have migrated, on May 8 the chain forked to new rules, the devs did a massive revocation of old tickets and all revocation code/UI was removed from software. We should really have no "legacy tickets" to talk about at this point.

Personally I found the `VSP2` and `VSP3` qualifiers quite handy in my docs/notes, but perhaps changing old docs is unnecessary atm. We can still use them in new docs, or revisit this issue when we get another major protocol change and need `VSP3` and `VSP4` to disambiguate their tickets/servers/fees/UIs/etc.

Feel free to close.
